syntax = "proto3";

package stock.v1;

service Stock {
  rpc add_product(NewProductParams) returns (NewProductResponse) {}
  rpc update_product_quantity(UpdateProductParams) returns (UpdateProductResponse) {}
  rpc list_products(ListProductsParams) returns (ListProductsResponse) {}
  rpc kill_server(KillServerParams) returns (KillServerResponse) {}
}

message NewProductParams {
  string description = 1;
  int32 quantity = 2;
}

message NewProductResponse {
  int32 id = 1;
}

message UpdateProductParams {
  int32 id = 1;
  int32 value = 2;
}

message UpdateProductResponse {
  int32 status = 1;
}

message ListProductsParams {}

message ListProductsResponse {
  repeated Product products = 1;
}

message Product {
  int32 id = 1;
  int32 quantity = 2;
  string description = 3;
}

message KillServerParams {}

message KillServerResponse {
  int32 quantity = 1;
}
